src/main/scala/cromwell/engine/package.scala

```diff
@@ -26,6 +32,41 @@ package object engine {
     def randomId() = WorkflowId(UUID.randomUUID())
   }
 
+  /**
+   * Constructs a representation of a particular workflow invocation.  As with other
+   * case classes and apply() methods, this will throw an exception if it cannot be
+   * created
+   */
+  case class WorkflowDescriptor(id: WorkflowId, sourceFiles: WorkflowSourceFiles) {
+    val workflowOptions = Try(sourceFiles.workflowOptionsJson.parseJson) match {
+      case Success(options: JsObject) => WorkflowOptions.fromJsonObject(options).get // .get here to purposefully throw the exception
+      case _ => throw new Throwable(s"Workflow ${id.toString} contains bad workflow options JSON: ${sourceFiles.workflowOptionsJson}")
```

Comment:
The error message from the parse possibly had a better error message than just handing back the whole JSON, but the message has been lost with the `case _`